cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -715,7 +726,11 @@ private void doAddSchedule(HttpRequest request, HttpResponder responder, String
     List<Constraint> constraints = Objects.firstNonNull(scheduleFromRequest.getConstraints(), NO_CONSTRAINTS);
     ProgramSchedule schedule = new ProgramSchedule(scheduleName, description, programId, properties,
                                                    scheduleFromRequest.getTrigger(), constraints);
-    programScheduler.addSchedule(schedule);
+    try {
+      programScheduler.addSchedule(schedule);
+    } catch (IllegalArgumentException e) {
+      throw new BadRequestException(e);
```

Comment:
You're doing this translation in other parts of this PR also.
If possible, we should throw BadRequestException, instead of IllegalArgumentException at the root cause (I know its difficult for the TimeScheduler part).